devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -1769,140 +1828,30 @@ bool WholeBodyDynamicsDevice::attachAllVirtualAnalogSensor(const PolyDriverList&
-            if (tempName==ftName){
-                individualCheck++;
-                ftMap=ft;
-            }
-        }
-        if (individualCheck!=1){
-            yWarning()<< "wholeBodyDynamics : A temperature sensor name in multipleAnalogRemappedDevice do not match the name of the ft sensors";
-            yDebug()<<"wholeBodyDynamics : Could not find or found multiple times sensor "<<tempName<<" among ft sensor names";
-        }
-        else {
-            checkCounter++;
-            // assuming the name is the same the order withing the mas interfaces might have a different id so we need a mapping
-            ftTempMapping[static_cast<size_t>(ftMap)]=tSensor;
-            yInfo()<< "wholeBodyDynamics: ftTempMapping "<< ftMap << " is ft position in model "<< tSensor <<"="<<ftTempMapping[ftMap] << " sensor name "<< tempName;
-        }
-    }
-    if (checkCounter!=tempSensors){
-        yError("wholeBodyDynamics : Not all temperature sensors have the same name as the ft sensors, expected %d , found %d",tempSensors,checkCounter);
-        return false;
-    }
-    //End of temporary temperature check
-
-    // iterate through ftMultipleAnalogSensorNames
-    // check if each name is a FT sensor in the URDF
-    auto nrFTsInURDF = estimator.model().sensors().getNrOfSensors(iDynTree::SIX_AXIS_FORCE_TORQUE);
-    for(size_t IDTsensIdx=0; IDTsensIdx < nrFTsInURDF; IDTsensIdx++)
-    {
-        std::string sensorName = estimator.model().sensors().getSensor(iDynTree::SIX_AXIS_FORCE_TORQUE,IDTsensIdx)->getName();
-
-        bool ftInAnalog = (std::find(ftAnalogSensorNames.begin(), ftAnalogSensorNames.end(), sensorName) != ftAnalogSensorNames.end());
-        bool ftInMAS = (std::find(ftMultipleAnalogSensorNames.begin(), ftMultipleAnalogSensorNames.end(), sensorName) != ftMultipleAnalogSensorNames.end());
-
-        if (!ftInAnalog && !ftInMAS)
-        {
-            yError() << "WholeBodyDynamicsDevice was expecting a sensor named " << sensorName << " but it did not find one in the attached devices";
-            return false;
-        }
-    }
+    size_t nrOfFTSensorsInConfigFile = ftSensorNames.size();
 
     if (!settings.disableSensorReadCheckAtStartup) {
         // We try to read for a brief moment the sensors for two reasons:
         // so we can make sure that they actually work, and to make sure that the buffers are correctly initialized
         bool verbose = false;
-        double tic = yarp::os::Time::now();
-        bool timeSpentTryngToReadSensors = 0.0;
+        double timeSpentTryngToReadSensors = 0.0;
         bool readSuccessfull = false;
+        double tic = yarp::os::Time::now();
         while( (timeSpentTryngToReadSensors < wholeBodyDynamics_sensorTimeoutInSeconds) && !readSuccessfull )
         {
             readSuccessfull = readFTSensors(verbose);
             timeSpentTryngToReadSensors = (yarp::os::Time::now() - tic);
+            yarp::os::Time::delay(0.001);
```

Comment:
Why is this needed?

Reply:
Due to a bug, I actually ended up having a sensor check that was failing, and I realized that this part of code was extremely bugged (we were storing time in a bool : ) ). One bug that if the sensor read failed, we continued to try to read from the sensor without any kind of wait, using ~100% of the CPU. I added the delay to fix this problem. However, looking at the code now, it does not make sense to wait 1 ms if the sensor read was successful, let me change that.

Reply:
Fixed in https://github.com/robotology/whole-body-estimators/pull/194/commits/a67d28734036156f0729c537771a20338cc38803 .